frontend/src/stores/roles/actions.ts

```diff
@@ -38,4 +39,9 @@ async function removeRole(roleId: string) {
   roles.value = roles.value.filter((role) => role.id !== parseInt(roleId))
 }
 
-export { fetchRoles, addRole, editRoleAction, removeRole }
+async function addLevel(level: Level, role: Role): Promise<void> {
+  await post<Level[]>('/rolelevel', level)
+  role.levels?.push(level)
```

Comment:
Here you are just pushing a level to a random array. The point is that you should push a level to the respective Role, agree?